### PR TITLE
Refactor linop testing using pytest-cases

### PR DESCRIPTION
**Pull request by marvinpfoertner**
This PR refactors the current tests (excluding arithmetic tests) for linear operators to use pytest and pytest-cases.
It also makes existing tests stricter and adds new tests.

### CODE REVIEW

File: src/probnum/linops/_linear_operator.py [hunk 1]
```diff
@@ -229,19 +249,18 @@ def trace(self):
...
-            _identity = np.eye(self.shape[0])
-            trace = 0.0
-            for i in range(self.shape[0]):
-                trace += np.squeeze(
-                    _identity[np.newaxis, i, :]
-                    @ self.matvec(_identity[i, :, np.newaxis])
-                )
-            return trace
+        if not self.is_square:
+            raise np.linalg.LinAlgError("The trace is only defined on square operators")
```
Comment (JonathanWenger) [inline src/probnum/linops/_linear_operator.py:255]: Petty: Full stop missing.

File: tests/test_linops/test_linops.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        np.testing.assert_allclose(linop_eigvals, matrix_eigvals)
+    else:
+        with pytest.raises(np.linalg.LinAlgError):
+            linop.eigvals()
+
+
+@pytest.mark.parametrize("p", [None, 2])
+@pytest_cases.parametrize_with_cases("linop,matrix", cases=case_modules)
+def test_cond(linop: pn.linops.LinearOperator, matrix: np.ndarray, p: Union[None, int]):
+    linop_det = linop.cond(p=p)
```
Comment (JonathanWenger) [inline tests/test_linops/test_linops.py:190]: naming should be `linop_cond` and `matrix_cond`. This seems like a copy-paste error.

**Review by JonathanWenger — APPROVED** [review]
Great job! The pytest-cases addition will be very useful to simplify testing in the entire package.

File: tests/test_linops/test_linops.py (hunk 2 above)
Comment (marvinpfoertner) [reply tests/test_linops/test_linops.py]: You got me there

**Pull request merged**